app/controllers/venues_controller.rb

```diff
@@ -1,8 +1,7 @@
 class VenuesController < ApplicationController
   def show
     @venue = Venue.find_by(id: params[:id])
-    possible_event_ids = Event.where(venue_id: @venue.id).uniq.map(&:id)
-    @items = Item.available.active.where.not(id: session[:cart]).where(event_id: possible_event_ids)
+    @items = Item.active.not_in_cart(session[:cart]).joins(:event).where(events: {venue_id: @venue.id})
     @events = @items.map(&:event).uniq
```

Comment:
finds, all the items that are active and not in cart, and then have an event which a specific venue.